react-training/src/app/components/App/index.js

```diff
@@ -1,16 +1,40 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { connect } from 'react-redux';
 
+import { authActions } from '../../../redux/auth/action';
 import Login from '../../screens/Login';
 import Game from '../../screens/Game';
+import PrivateRoute from '../PrivateRoute';
 
-const App = () => (
-  <Router>
-    <Switch>
-      <Route exact path="/" component={Login} />
-      <Route path="/game" component={Game} />
-    </Switch>
-  </Router>
-);
+class App extends Component {
+  componentDidMount() {
+    this.props.setAuthentication();
+  }
 
-export default App;
+  render() {
+    return (
+      <Router>
+        <Switch>
+          <Route path="/login" component={Login} />
+          {['/', 'game'].map(route => (
+            <PrivateRoute key={route} path={route} component={Game} />
```

Comment:
Move all auth logic to `PrivateRoute`, then rename the component  to `AuthenticatedRoute` or something more descriptible. U must use this `AuthenticatedRoute` component for all routes defined in this component